### PR TITLE
TRAFODION - 2632 Performing update statistics on metadata tables caus…

### PR DESCRIPTION
**Pull request by robertamarton**
…es issues

Added code to return error and cleanup code a bit:
*** ERROR[9205] UPDATE STATISTICS is not supported for object <metadata table>

Also fixed a problem where privileges on Hive tables that included external
table gave false "no priv" error for the SHOWDDL <hive table> command.

### CODE REVIEW

File: core/sql/ustat/hs_globals.cpp [hunk 1]
```diff
@@ -3776,7 +3776,7 @@ NABoolean HSGlobalsClass::isAuthorized(NABoolean isShowStats)
        if (privs == NULL)
          {
            *CmpCommon::diags() << DgSqlCode(-1034);
-            authorized = FALSE;
+            return FALSE;
```
Comment (DaveBirdsall) [inline core/sql/ustat/hs_globals.cpp]: Changing this to an immediate return breaks the timer at line 3795. Please either add a call to LM->LogTimeDiff or revert this change.

File: core/sql/ustat/hs_globals.cpp [hunk 2]
```diff
@@ -3788,7 +3788,7 @@ NABoolean HSGlobalsClass::isAuthorized(NABoolean isShowStats)
             << DgSqlCode( -4481 )
             << DgString0( "SELECT or MANAGE_STATISTICS" )
             << DgString1( objDef->getNATable()->getTableName().getQualifiedNameAsAnsiString() );
-            authorized = FALSE;
+            return FALSE;
```
Comment (DaveBirdsall) [inline core/sql/ustat/hs_globals.cpp]: Ditto

File: core/sql/ustat/hs_parser.cpp [hunk 3]
```diff
@@ -178,89 +178,24 @@ Lng32 AddTableName( const hs_table_type type
...
-            if (SqlParser_NAMETYPE == DF_NSK)
-              {
-                catName = SqlParser_MPLOC.getSysDotVol();
-                hs_globals->tableType = GUARDIAN_TABLE;
-                hs_globals->tableFormat = SQLMP;
-              }
-            else
-            // LCOV_EXCL_STOP
-              catName = ActiveSchemaDB()->getDefaultSchema().getCatalogName();
+            catName = ActiveSchemaDB()->getDefaultSchema().getCatalogName();
```
Comment (DaveBirdsall) [inline core/sql/ustat/hs_parser.cpp]: Thank you for getting rid of the NSK anachronisms.

File: core/sql/ustat/hs_globals.cpp (hunk 1 above)
Comment (robertamarton) [inline core/sql/ustat/hs_globals.cpp]: The return was added because the next statement "if ( privs->hasSelectPriv() )" cores if privs is NULL.     But will change the code to not break the time.
